main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -442,12 +443,50 @@ object Verifier {
       val t = visitExpr(exp)
       checkEq(tpe, t, loc)
 
-    case Expr.TryWith(_, _, _, _, tpe, _, _) =>
-      // TODO: VERIFIER: Add support for TryWith.
-      tpe
+    case Expr.TryWith(exp, effUse, rules, ct, tpe, purity, loc) =>
+      val exptype = visitExpr(exp) match {
+        case MonoType.Arrow(List(u), t) =>
+          check(expected = MonoType.Unit)(actual = u, loc)
+          t
+        case e => failMismatchedShape(e, "Arrow with one parameter", loc)
```

Comment:
I think these errors needs to be normalized/refactored at some point but for now its fine

Reply:
yea, i'll do it in a seperate PR.